copyparty/web/browser.js

```diff
@@ -9393,6 +9447,198 @@ ebi('files').onclick = ebi('docul').onclick = function (e) {
 };
 
 
+
+let rcm = (function () {
+	if (MOBILE)
+		return {enabled: false}
+
+	let r = {
+		enabled: true,
+		double: true
+	};
+	bcfg_bind(r, 'enabled', 'ren', true);
+	bcfg_bind(r, 'double', 'rno', true);
+
+	let menu = ebi('rcm');
+	let selFile = {
+		elem: null,
+		type: null,
+		path: null,
+		id: null,
+		relpath: null,
+		no_dsel: false
+	};
+
+	function mktemp(is_dir) {
+		let row = mknod('tr', 'temp', 
+			'<td>-new-</td>' +
+			'<td columnspan="2"><input id="tempname" class="i" type="text" placeholder="' + (is_dir ? 'Folder' : "File") + ' Name"></td>'
+		);
+		QS("#files tbody").appendChild(row);
+
+		function sendit(name) {
+			name = ('' + name).trim();
+			if (!name)
+				return;
+			let data = new FormData();
+			data.set("act", is_dir ? "mkdir" : "new_md");
+			data.set("name", name);
+
+			let req = new XHR();
+			req.open("POST", get_evpath());
+			req.onload = req.onerror = function() {
+				if (req.status != 201)
+					return toast.err(3, "something broke :(");
+				toast.ok(3, "OK :)");
+				reload_browser();
+			};
+			req.send(data);
+		}
+
+		let input = ebi("tempname");
+		input.onblur = function() {
+			sendit(input.value);
+			row.remove();
+		};
+		input.onkeydown = function(e) {
+			if (e.key == "Enter")
+				sendit(input.value);
+			if (e.key == "Enter" || e.key == "Escape")
```

Comment:
pressing escape to cancel creating a new file also closes the top menu (up2k/settings/etc)